iroha/src/block_sync.rs

```diff
@@ -163,21 +168,24 @@ pub mod message {
 
     impl VersionedMessage {
         /// Same as [`as_v1`] but also does conversion
+        #[inline]
         pub const fn as_inner_v1(&self) -> &Message {
-            match self {
-                Self::V1(v1) => &v1.0,
+            match *self {
+                Self::V1(ref v1) => &v1.0,
             }
         }
 
         /// Same as [`as_inner_v1`] but returns mutable reference
+        #[inline]
         pub fn as_mut_inner_v1(&mut self) -> &mut Message {
-            match self {
-                Self::V1(v1) => &mut v1.0,
+            match *self {
+                Self::V1(ref mut v1) => &mut v1.0,
```

Comment:
We do we need to dereference here and then capture by reference? Previous syntax is simpler and more readable.

Reply:
@i1i1 same for all the other similar enum cases that you corrected